x-pack/plugins/apm/typings/elasticsearch.ts

```diff
@@ -6,7 +6,7 @@
 
 import { StringMap } from 'x-pack/plugins/apm/typings/common';
 
-declare module 'elasticsearch' {
+declare module '@elastic/elasticsearch' {
   // extending SearchResponse to be able to have typed aggregations
   export interface AggregationSearchResponse<T, U = void>
     extends SearchResponse<T> {
```

Comment:
@delvedor Currently I'm extending `SearchResponse` (which is now called `RequestParams.Search`) with a generic argument for aggregations. Is that part of the official types now, so maybe I can delete this??

Reply:
`RequestParams.Search` exposes the request parameters, as you can see [here](https://github.com/elastic/elasticsearch-js/blob/8e26dc874f51f1e1a3b39b363948fc3366023260/api/requestParams.d.ts#L1044-L1086) and not the search response. The search response is defined in kibana [here](https://github.com/delvedor/kibana/blob/next-es-client/typings/%40elastic/elasticsearch/index.d.ts).

The client does not offer the typings for request bodies and responses because there is not an official spec to generate the types from.